src/libs/antares/study/parts/hydro/finallevelinflowsmodifyer.cpp

```diff
@@ -1,187 +0,0 @@
-{
-    // calculate yearly inflows
-    auto const& srcinflows = hydro_.series->storage.getColumn(year);
-
-    double totalYearInflows = 0.0;
-    for (unsigned int day = 0; day < DAYS_PER_YEAR; ++day)
-        totalYearInflows += srcinflows[day];
-    return totalYearInflows;
-}
-
-bool FinalLevelInflowsModifier::preCheckStartAndEndSim(unsigned int year) const
-{
-    
-    int initReservoirLvlMonth = hydro_.initializeReservoirLevelDate; // month [0-11]
-    if (lastSimulationDay_ == DAYS_PER_YEAR && initReservoirLvlMonth == 0)
-        return true;
-
-    logs.error() << "Year: " << year + 1 << ". Area: " << areaName_
-                    << ". Simulation must end on day 365 and reservoir level must be "
-                    "initiated in January";
-    return false;
-}
-
-bool FinalLevelInflowsModifier::preCheckYearlyInflow(double totalYearInflows, unsigned int year) const
-{
-    double reservoirCapacity = hydro_.reservoirCapacity;
-    if (-deltaReservoirLevel_ * reservoirCapacity > totalYearInflows)
-    {
-        logs.error() << "Year: " << year + 1 << ". Area: " << areaName_
-                     << ". Incompatible total inflows: " << totalYearInflows
-                     << " with initial: " << initialReservoirLevel_
-                     << " and final: " << finalReservoirLevel_ << " reservoir levels.";
-        return false;
-    }
-    return true;
-}
-
-bool FinalLevelInflowsModifier::preCheckRuleCurves(unsigned int year) const
-{
-    double lowLevelLastDay  = hydro_.reservoirLevel[Data::PartHydro::minimum][DAYS_PER_YEAR - 1];
-    double highLevelLastDay = hydro_.reservoirLevel[Data::PartHydro::maximum][DAYS_PER_YEAR - 1];
-
-    if (finalReservoirLevel_ < lowLevelLastDay || finalReservoirLevel_ > highLevelLastDay)
-    {
-        logs.error() << "Year: " << year + 1 << ". Area: " << areaName_
-                     << ". Specifed final reservoir level: " << finalReservoirLevel_
-                     << " is incompatible with reservoir level rule curve [" << lowLevelLastDay
-                     << " , " << highLevelLastDay << "]";
-        return false;
-    }
-    return true;
-}
-
-void FinalLevelInflowsModifier::initialize(const Matrix<double>& scenarioInitialHydroLevels,
-                                           const Matrix<double>& scenarioFinalHydroLevels,
-                                           const unsigned int lastSimulationDay,
-                                           const unsigned int nbYears)
-{
-    isApplicable_.assign(nbYears, false);
-    deltaLevel.assign(nbYears, 0.);
-    InitialLevels_ = &(scenarioInitialHydroLevels.entry[areaIndex_]);
-    FinalLevels_ = &(scenarioFinalHydroLevels.entry[areaIndex_]);
-    lastSimulationDay_ = lastSimulationDay;
-}
-
-bool FinalLevelInflowsModifier::isActive()
-{
-    return  hydro_.reservoirManagement && 
-            !hydro_.useWaterValue &&
-            isValidLevel(finalReservoirLevel_) &&
-            isValidLevel(initialReservoirLevel_);
```

Comment:
The initial level is no more tested.
Why ? Because it cannot have an invalid value.
Indeed, either : 
- it is fixed by user in the scenario builder
- or it is computed randomly between bound (0 and 100).